core-blocks/button/edit.js

```diff
@@ -106,22 +114,35 @@ class ButtonEdit extends Component {
 						keepPlaceholderOnFocus
 					/>
 					<InspectorControls>
-						<PanelColor
-							colorValue={ backgroundColor.value }
-							title={ __( 'Background Color' ) }
-							onChange={ setBackgroundColor }
-						/>
-						<PanelColor
-							colorValue={ textColor.value }
-							title={ __( 'Text Color' ) }
-							onChange={ setTextColor }
-						/>
-						{ this.nodeRef && <ContrastCheckerWithFallbackStyles
-							node={ this.nodeRef }
-							textColor={ textColor.value }
-							backgroundColor={ backgroundColor.value }
-							isLargeText={ true }
-						/> }
+						<PanelColorSettings
+							title={ __( 'Color Settings' ) }
+							colorSettings={ [
```

Comment:
To note, The only reason we are passing the color settings as an array of objects instead of using directly a React component is because if we pass as settings object we can use the same values for the color indicators and avoid passing repeated information. Another approach would be access child components of PanelColorSettings from PanelColorSettings which seems an anti-pattern or using slot-fill which seems complex for this case. Maybe we can add some comments to PanelColorSettings explaining this. 
I like this version.